experimental/origin-dapp2/src/pages/growth/Campaigns.js

```diff
@@ -0,0 +1,615 @@
+    return (
+      <div className="reward d-flex mr-4 align-items-center pl-2 pt-2 pb-2 mt-2">
+        <img src="images/ogn-icon.svg" />
+        <div className="value">
+          {renderPlusSign ? '+' : ''}
+          {formatTokens(amount)}
+        </div>
+      </div>
+    )
+  }
+
+  return (
+    <div className="d-flex action">
+      <div className="col-2 d-flex justify-content-center">
+        <div className="image-holder mt-auto mb-auto">
+          {type !== 'referral' && (
+            <Fragment>
+              <img className="background" src={backgroundImgSrc} />
+              <img className={type} src={foregroundImgSrc} />
+            </Fragment>
+          )}
+          {type === 'referral' && (
+            <img className="astronaut" src="images/growth/astronaut-icon.svg" />
+          )}
+          {showLockIcon && (
+            <img className="lock" src="images/growth/lock-icon.svg" />
+          )}
+        </div>
+      </div>
+      <div className="col-8 d-flex flex-column">
+        <div className="title">{title}</div>
+        <div className="info-text">{infoText}</div>
+        <div className="d-flex">
+          {type === 'referral' &&
+            rewardPending !== null &&
+            rewardPending.amount > 0 && (
+              <Fragment>
+                <div className="d-flex align-items-center sub-text">
+                  Pending
+                </div>
+                {renderReward(rewardPending.amount, true)}
+              </Fragment>
+            )}
+          {actionCompleted && rewardEarned !== null && (
+            <Fragment>
+              <div className="d-flex align-items-center sub-text">Earned</div>
+              {renderReward(rewardEarned.amount, false)}
+            </Fragment>
+          )}
+          {!actionCompleted &&
+            reward !== null &&
+            renderReward(reward.amount, true)}
+        </div>
+      </div>
+      <div className="col-2 d-flex">
+        {!actionCompleted && (
+          <Link to="/profile" className="mt-auto mb-auto">
+            <button
+              className="btn btn-primary btn-rounded mr-2"
+              children="Go"
+            />
+          </Link>
+        )}
+      </div>
+    </div>
+  )
+}
+
+function ActionList(props) {
+  return (
+    <Fragment>
+      <div className="d-flex flex-column">
+        {props.title !== undefined && (
+          <div className="action-title">{props.title}</div>
+        )}
+        {props.actions.map(action => {
+          return (
+            <Action
+              action={action}
+              decimalDevision={props.decimalDevision}
+              key={`${action.type}:${action.status}`}
+            />
+          )
+        })}
+      </div>
+    </Fragment>
+  )
+}
+
+function Campaign(props) {
+  const { campaign, accountId } = props
+  const { startDate, endDate, status, rewardEarned, actions, name } = campaign
+
+  const formatTimeDifference = (start, end) => {
```

Comment:
Perhaps we can move this to `utils/formatTimeDifference` - there's already a `utils/distanceToNow.js`

Reply:
Good idea, doing that tnx